### PR TITLE
tls: Update README

### PR DESCRIPTION
**Pull request by martinpitt**
We forgot to keep this file updated with the recent work of  running
cockpit-ws as different user, and adding client certificate
authentication.

### CODE REVIEW

File: src/tls/README.md [hunk 1]
```diff
@@ -9,10 +9,9 @@ multiplexes sessions of different users in one process. Thus any vulnerability
 can easily lead to complete privilege escalation across all running and future
 sessions.
 
-Right now this does not have any extra features over using cockpit-ws directly
-(other than better isolation). In the future, it will be able to do certificate
-based client authentication (smart cards, browser-imported certificates, and
-similar), which greatly aggravates the above security trust issue.
+That separation permits Cockpit to do certificate based client authentication
+(smart cards, browser-imported certificates, and similar) in a sufficiently
+robust and trustworthy manner.
```
Comment (allisonkarlitskaya) [inline src/tls/README.md]: I feel like this is a bit too hand-wavy.  I'd specifically mention that using cockpit-tls allows us to isolate the main parts of cockpit on a per-client-certificate basis.

File: src/tls/README.md [hunk 2]
```diff
@@ -29,40 +28,88 @@ Design goals
...
+   - If not, it connects to [cockpit-wsinstance-http.socket](../src/ws/cockpit-wsinstance-http.socket.in) or
+     [cockpit-wsinstance-http-redirect.socket](../src/ws/cockpit-wsinstance-http-redirect.socket.in).
+   - If the connection does use TLS, it calculates the fingerprint of the
+     client certificate (using the empty string if there is none), and connects
+     to [cockpit-wsinstance-https-factory.socket](../src/ws/cockpit-wsinstance-https-factory.socket.in).
+     This starts a helper factory process `cockpit-wsinstance-factory` that
+     reads the fingerprint from stdin, and asks systemd to start a new
+     [cockpit-wsinstance-https@fingerprint.socket](../src/ws/cockpit-wsinstance-https@.socket.in)
+     and .service pair.
+ * All these instances run in their own systemd cgroup, as another unprivileged
```
Comment (allisonkarlitskaya) [inline src/tls/README.md]: "Each of these instances runs in its own..." would make it more clear that there are many cgroups.

File: src/tls/README.md (hunk 1 above)
Comment (martinpitt) [reply src/tls/README.md]: Right, I reworked the text.

File: src/tls/README.md (hunk 2 above)
Comment (martinpitt) [reply src/tls/README.md]: Thanks for the suggestion, that was indeed ambiguous. Updated.

File: src/tls/README.md [hunk 3]
```diff
@@ -29,40 +31,88 @@ Design goals
...
+   - If not, it connects to [cockpit-wsinstance-http.socket](../src/ws/cockpit-wsinstance-http.socket.in) or
+     [cockpit-wsinstance-http-redirect.socket](../src/ws/cockpit-wsinstance-http-redirect.socket.in).
+   - If the connection does use TLS, it calculates the fingerprint of the
+     client certificate (using the empty string if there is none), and connects
+     to [cockpit-wsinstance-https-factory.socket](../src/ws/cockpit-wsinstance-https-factory.socket.in).
+     This starts a helper factory process `cockpit-wsinstance-factory` that
+     reads the fingerprint from stdin, and asks systemd to start a new
+     [cockpit-wsinstance-https@fingerprint.socket](../src/ws/cockpit-wsinstance-https@.socket.in)
+     and .service pair.
+ * Each of these instances run in a separate systemd cgroup, as another
```
Comment (allisonkarlitskaya) [inline src/tls/README.md]: s/run/runs/ now that it's "each of" (which is singular).

I also really liked "its own" as the thing which added the emphasis.  Is there a reason that you didn't go with that?

Comment (martinpitt) [reply src/tls/README.md]: Updated again, thanks!

**Pull request merged**